gitpod-io/gitbot
# plugins/deployed-labeler: New plugin that labels deployed PRs

## Description

It receives "commit" and "team" as parameters, then adds the label `deployed` to all pull requests(that has the label `team: <team-parameter>`) associated with a merged commit in the commit tree.

## Related Issue(s)

Fixes https://github.com/gitpod-io/ops/issues/159

## How to test

Retrieve hmac and github token from our Prow cluster and run `main.go`  with:
```
go run main.go -hmac=hmac --github-token-path=token
```


Then on a new terminal run the command: 
```
curl -XPOST 'localhost:8080/deployed?commit=0a483885ff9a6ff1149c0388d1751635fd8cd788&team=test'
```

It should return 2 arrays, where the first is the URL of all pull requests that were successfully labeled and the second is an array of errors that may have occurred in the process.

E.g.:
![image](https://user-images.githubusercontent.com/24193764/139126959-cffd9768-700d-453a-b9f8-e8862154cacd.png)


## Release Notes

```release-note
NONE
```

## Documentation
